app/src/main/java/org/fossasia/openevent/general/auth/RequestAuthenticator.kt

```diff
@@ -1,18 +1,19 @@
 package org.fossasia.openevent.general.auth
 
-import okhttp3.Authenticator
-import okhttp3.Request
+import okhttp3.Interceptor
 import okhttp3.Response
-import okhttp3.Route
 
-class RequestAuthenticator(private val authHolder: AuthHolder) : Authenticator {
+class RequestAuthenticator(private val authHolder: AuthHolder) : Interceptor {
 
-    override fun authenticate(route: Route?, response: Response): Request? {
-        val authorization: String? = authHolder.getAuthorization()
-        return if (response.request().header("Authorization") == null && authorization != null) {
-            response.request().newBuilder()
-                    .header("Authorization", authorization)
-                    .build()
-        } else null // Give up, we already tried authenticating
+    override fun intercept(chain: Interceptor.Chain): Response? {
+        val authorization = authHolder.getAuthorization()
+        val original = chain.request()
+        val request = original.newBuilder()
+            .header("Authorization", authorization.toString())
```

Comment:
Why toString?

Reply:
It is showing error while running `Authorization == null` and network module is not working. Because here `authorization` type is nullable String.

Reply:
Only run this if (!Utils.isEmpty), problem solved

Reply:
Updated